kuma/wiki/search.py

```diff
@@ -0,0 +1,255 @@
+                    ],
+                },
+                'kuma_title': {
+                    'type': 'custom',
+                    'tokenizer': 'standard',
+                    'filter': [
+                        'elision',
+                        'kuma_word_delimiter',
+                        'lowercase',
+                        'standard',
+                        'snowball',
+                    ],
+                },
+                'case_sensitive': {
+                    'type': 'custom',
+                    'tokenizer': 'keyword'
+                },
+                'case_insensitive_keyword': {
+                    'type': 'custom',
+                    'tokenizer': 'keyword',
+                    'filter': 'lowercase'
+                }
+            },
+        }
+
+    @classmethod
+    def get_settings(cls):
+        settings = {
+            'mappings': cls.get_mapping(),
+            'settings': {
+                'index': {
+                    'analysis': cls.get_analysis()
+                }
+            }
+        }
+
+        return settings
+
+    @classmethod
+    def bulk_index(cls, documents, id_field='id', es=None, index=None):
+        """Index of a bunch of documents."""
+        es = es or cls.get_connection()
+        index = index or cls.get_index()
+        type = cls.get_doc_type()
+
+        actions = [
+            {'_index': index, '_type': type, '_id': d['id'], '_source': d}
+            for d in documents]
+
+        bulk(es, actions)
+
+    @classmethod
+    def get_index(cls):
+        from kuma.search.models import Index
+        return Index.objects.get_current().prefixed_name
+
+    @classmethod
+    def search(cls, **kwargs):
+        kwargs.update({
+            'using': connections.get_connection(),
+            'index': cls.get_index(),
+            'doc_type': {cls._doc_type.name: cls.from_es},
+        })
+        sq = Search(**kwargs)
+
+        # Add highlighting.
+        sq = sq.highlight(*cls.excerpt_fields)
+        sq = sq.highlight_options(order='score')
+
+        return sq
+
+    @classmethod
+    def get_model(cls):
+        from kuma.wiki.models import Document
+        return Document
+
+    @classmethod
+    def get_indexable(cls, percent=100):
+        """
+        For this mapping type return a list of model IDs that should be
+        indexed with the management command, in a full reindex.
+
+        WARNING: When changing this code make sure to update the
+                 ``should_update`` method below, too!
+
+        """
+        model = cls.get_model()
+
+        excludes = []
+        for exclude in cls.exclude_slugs:
+            excludes.append(Q(slug__icontains=exclude))
+
+        qs = (model.objects
+                   .filter(is_template=False,
+                           is_redirect=False,
+                           deleted=False)
+                   .exclude(reduce(operator.or_, excludes)))
+
+        percent = float(percent) / 100
```

Comment:
We should use the new division behavior here to prepare ourselves for future Python version. So add `from __future__ import division` and use `//`